src/gui/guiButtonKey.h

```diff
@@ -0,0 +1,80 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include "guiButton.h"
+#include "client/keycode.h"
+#include "util/string.h"
+#include "gettext.h"
+
+using namespace irr;
+
+class GUIButtonKey : public GUIButton
+{
+	using super = GUIButton;
+
+public:
+	//! Constructor
+	GUIButtonKey(gui::IGUIEnvironment *environment, gui::IGUIElement *parent,
+			s32 id, core::rect<s32> rectangle, ISimpleTextureSource *tsrc,
+			bool noclip = false)
+		: GUIButton(environment, parent, id, rectangle, tsrc, noclip) {}
+
+	//! Sets the text for the key field
+	virtual void setText(const wchar_t *text) override
+	{
+		setKey(wide_to_utf8(text));
+	}
+
+	//! Gets the value for the key field
+	virtual const wchar_t *getText() const override
+	{
+		return keysym.c_str();
+	}
+
+	//! Do not drop returned handle
+	static GUIButtonKey *addButton(gui::IGUIEnvironment *environment,
+			const core::rect<s32> &rectangle, ISimpleTextureSource *tsrc,
+			IGUIElement *parent, s32 id, const wchar_t *text = L"",
+			const wchar_t *tooltiptext = L"");
+
+	//! Called if an event happened
+	virtual bool OnEvent(const SEvent &event) override;
+
+	//! Start key capture
+	void startCapture()
+	{
+		capturing = true;
+		super::setText(wstrgettext("Press Key").c_str());
+	}
+
+	//! Cancel key capture
+	void cancelCapture()
+	{
+		capturing = false;
+		super::setText(wstrgettext(key_value.name()).c_str());
+	}
+
+	//! Check whether the field is capturing a key
+	bool isCapturing() const
+	{
+		return capturing;
+	}
+
+	//! Sets the captured key and stop capturing
+	void setKey(KeyPress key);
+
+	//! Gets the captured key
+	KeyPress getKey() const
+	{
+		return key_value;
+	}
+
+private:
+	void sendKey();
+
+	bool capturing = false;
+	KeyPress key_value = {};
+	std::wstring keysym;
```

Comment:
I would prefer to use the `m_ ` prefix for these variables. I find it very helpful to see the scope of a variable within a function without having to rely on IDE features (which might also require interactions to reveal this information).

Although it seems that the code style guidelines make it seem optional? In this case I'll leave the decision to you.